lib/cn/route.c

```diff
@@ -52,19 +53,42 @@ route_node_key_set(
         uint fmtarg = (nodeoff + 1) * map->rtm_skip - 1;
 
         if (map->rtm_fmt) {
-            int n = snprintf((char *)node->rtn_keybuf, sizeof(node->rtn_keybuf), map->rtm_fmt,
-                             fmtarg);
-
-            if (n < 1 || n >= sizeof(node->rtn_keybuf)) {
-                log_err("overflow %u: n %d, pfxlen %u, fmt [%s], fmtarg %u",
-                        nodeoff, n, pfxlen, map->rtm_fmt, fmtarg);
-                abort();
-            }
-
-            if (n != pfxlen) {
-                log_err("skipping %u: n %d, pfxlen %u, fmt [%s], fmtarg %u",
-                        nodeoff, n, pfxlen, map->rtm_fmt, fmtarg);
-                return -1;
+            if (!strcmp(map->rtm_fmt, "MainKvs")) {
```

Comment:
Does this work for any number of collections?

Reply:
No, if you have more than one collection (e.g., sysbench) you need to use the "binary" option to key on the collection ID for distribution across nodes.  This is just a temporary hack for imp and ycsb over mongod...